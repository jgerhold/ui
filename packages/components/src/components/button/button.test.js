import { mount } from '@vue/test-utils'
import Button from './Button'

describe('Button', () => {
  describe('component', () => {
    test('should be a Vue instance', () => {
      const wrapper = mount(Button)
      expect(wrapper.isVueInstance()).toBeTruthy()
    })

    test('should provide defaults', async () => {
      const wrapper = await mount(Button, {
        slots: {
          default: '<span>Button Inner</span>'
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('props', () => {
    const props = [
      {
        prop: 'textColor',
        value: '#fff'
      },
      {
        prop: 'backgroundColor',
        value: '#fff'
      },
      {
        prop: 'borderColor',
        value: '#fff'
      },
      {
        prop: 'href',
        value: 'http://foo.bar'
      },
      {
        prop: 'disabled',
        value: true
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should have a property for ${prop}`, async () => {
        const wrapper = await mount(Button, {
          slots: {
            default: '<span>Button Inner</span>'
          },
          propsData: {
            [prop]: value
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
