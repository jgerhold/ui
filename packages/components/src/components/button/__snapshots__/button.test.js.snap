// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should provide defaults 1`] = `
<button
  class="button"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #2b8ac6;"
>
  <span
    class="inner centered"
  >
    <span>
      Button Inner
    </span>
  </span>
</button>
`;

exports[`Button props should have a property for backgroundColor 1`] = `
<button
  class="button"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #2b8ac6;"
>
  <span
    class="inner centered"
  >
    <span>
      Button Inner
    </span>
  </span>
</button>
`;

exports[`Button props should have a property for borderColor 1`] = `
<button
  class="button"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #fff;"
>
  <span
    class="inner centered"
  >
    <span>
      Button Inner
    </span>
  </span>
</button>
`;

exports[`Button props should have a property for disabled 1`] = `
<button
  class="button disabled"
  disabled="disabled"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #2b8ac6;"
>
  <span
    class="inner centered"
  >
    <span>
      Button Inner
    </span>
  </span>
</button>
`;

exports[`Button props should have a property for href 1`] = `
<a
  class="button"
  href="http://foo.bar"
  style="color: rgb(43, 138, 198); background: rgb(255, 255, 255); border-color: #2b8ac6;"
>
  <span
    class="inner centered"
  >
    <span>
      Button Inner
    </span>
  </span>
</a>
`;

exports[`Button props should have a property for textColor 1`] = `
<button
  class="button"
  style="color: rgb(255, 255, 255); background: rgb(255, 255, 255); border-color: #2b8ac6;"
>
  <span
    class="inner centered"
  >
    <span>
      Button Inner
    </span>
  </span>
</button>
`;
